### PR TITLE
Add swagger and api-docs endpoint

### PR DESCRIPTION
**Pull request by tsingh777**
image

### CODE REVIEW

File: swagger.json [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "searchtype",
+            "in": "query",
+            "description": "Type of search. { 0: First letter each word from start (Gurmukhi), 1: First letter each word anywhere(Gurmukhi), 2: Full Word (Gurmukhi), 3: Full Word Translation (English), 4: Romanized Gurmukhi (English), 5: Ang, 6: Main Letter (Gurmukhi), 7: Romanized first letter anywhere (English) }",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/SeaerchType"
```
Comment (navdeepsinghkhalsa) [inline swagger.json:44]: `SeaerchType` <- typo. Also in the definitions

File: swagger.json [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+            "name": "query",
+            "in": "path",
+            "description": "Search query",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "searchtype",
+            "in": "query",
+            "description": "Type of search. { 0: First letter each word from start (Gurmukhi), 1: First letter each word anywhere(Gurmukhi), 2: Full Word (Gurmukhi), 3: Full Word Translation (English), 4: Romanized Gurmukhi (English), 5: Ang, 6: Main Letter (Gurmukhi), 7: Romanized first letter anywhere (English) }",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:41]: Space before `(Gurmukhi)` in `1`?

File: swagger.json [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/SeaerchType"
+            }
+          },
+          {
+            "name": "source",
+            "in": "query",
+            "description": "Source to search in. { G: Guru Granth Sahib Ji, D: Dasam Granth Sahib, B: Bhai Gurdas Ji Vaaran, A: Amrit Keertan, S: Bhai Gurdas Singh Ji Vaaran N: Bhai Nand Lal Ji Vaaran, R: Rehatnamas & Panthic Sources }",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:50]: Comma needed after `S: Bhai Gurdas Singh Ji Vaaran`

File: swagger.json [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+            "name": "writer",
+            "in": "query",
+            "description": "Which writer to limit search to.",
+            "required": false,
+            "type": "number"
+          },
+          {
+            "name": "page",
+            "in": "query",
+            "description": "Pagination param. Which page to return.",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:80]: Though "Pagination param" is specified, there could be addition clarity with "Which results page to return"

File: swagger.json [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+          "200": {
+            "description": "successful operation"
+          }
+        }
+      }
+    },
+    "/shabads/{shabadID}": {
+      "get": {
+        "summary": "Shabads by ID",
+        "description": "Get an entire Shabad from its ShabadID. More than one shabadID can be specificed with operators '+' or ','",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:102]: `ShabadID` should be `shabadID`?

File: swagger.json [hunk 6]
```diff
@@ -0,0 +1,421 @@
...
+    },
+    "/angs/{pageNo}/{sourceID}": {
+      "get": {
+        "summary": "Ang",
+        "description": "Get an entire Ang(s) by number and optional source. More than one Ang can be specificed with operators '+' and '-'",
+        "parameters": [
+          {
+            "name": "pageNo",
+            "in": "path",
+            "description": "Ang number and/or range",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:130]: I don't think this takes both a number and range, so should this be "Ang number or range"?

File: swagger.json [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "sourceID",
+            "in": "path",
+            "description": "Source to search in. { G: Guru Granth Sahib Ji, D: Dasam Granth Sahib, B: Bhai Gurdas Ji Vaaran, A: Amrit Keertan, S: Bhai Gurdas Singh Ji Vaaran N: Bhai Nand Lal Ji Vaaran, R: Rehatnamas & Panthic Sources }",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:137]: Comma needed after `S: Bhai Gurdas Singh Ji Vaaran`

File: swagger.json [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+    },
+    "/hukamnamas/{year}/{month}/{day}": {
+      "get": {
+        "summary": "Hukamnama",
+        "description": "Get the Hukamnama from Sri Harmandir Sahib.",
+        "parameters": [
+          {
+            "name": "year",
+            "in": "path",
+            "description": "Year for specific date of hukamnama. Required if month and day is specified.",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:162]: Capitalized "Hukamnama" and "is" should be "are". Same for next two params

File: swagger.json [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+    },
+    "/random/{sourceID}": {
+      "get": {
+        "summary": "Retrieve a random Shabad",
+        "description": "Get Randomly selected  Shabad.",
+        "parameters": [
+          {
+            "name": "sourceID",
+            "in": "path",
+            "description": "Source to search in. { G: Guru Granth Sahib Ji, D: Dasam Granth Sahib, B: Bhai Gurdas Ji Vaaran, A: Amrit Keertan, S: Bhai Gurdas Singh Ji Vaaran N: Bhai Nand Lal Ji Vaaran, R: Rehatnamas & Panthic Sources }",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:199]: Comma after `S: Bhai Gurdas Singh Ji Vaaran`

File: swagger.json [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+          "200": {
+            "description": "successful operation"
+          }
+        }
+      }
+    },
+    "/banis/{baniID}": {
+      "get": {
+        "summary": "Bani",
+        "description": "Get a specific bani found in sundar gutka by BaniID.",
```
Comment (navdeepsinghkhalsa) [inline swagger.json:230]: `baniID` for consistency?

File: swagger.json (hunk 6 above)
Comment (tsingh777) [reply swagger.json:130]: 6+12-13 is possible. I'll include some examples.

**Review by navdeepsinghkhalsa — APPROVED** [review]

**Review by gurjit03 — APPROVED** [review]

**Pull request merged**